docs/api/routing.rst

```diff
@@ -35,6 +35,16 @@ A custom router is any class that implements the following interface:
                     the requested path
             """
 
+The `find` method of a custom router can optionally accept a `req` keyword argument
```

Comment:
I'd recommend instead of adding this extra section, just add a kwarg to the original interface definition above. If you like, you may also add a note about how this kwarg was added in 1.2 but routers that do not define it are still supported.